dbt/adapters/spark/impl.py

```diff
@@ -364,6 +364,24 @@ def get_rows_different_sql(
 
         return sql
 
+    # This is for use in the test suite
+    def run_sql_for_tests(self, sql, fetch, conn):
+        cursor = conn.handle.cursor()
+        try:
+            cursor.execute(sql)
+            if fetch == "one":
+                return cursor.fetchall()[0]
+            elif fetch == "all":
+                return cursor.fetchall()
```

Comment:
Looking at https://github.com/dbt-labs/dbt-core/pull/4986/commits/fd1caf410447940ab16a9fb682296b90aa134482#diff-89b13d4762b57109dfd58a03ba608e474e58df7fd112973498f64df235ca855eL230-L234:

We need all the `hasattr` checks in `dbt-spark`, right? To account for different connection methods having slightly different cursors

Reply:
The 'commit' and 'rollback' methods just log a NotImplemented if called, so there's not any point in calling them that I can see. Unless I'm missing something?